src/env0-environments-provider.ts

```diff
@@ -129,7 +126,8 @@ export class Env0EnvironmentsProvider
     Environment | undefined | null | void
   > = this._onDidChangeTreeData.event;
 
-  refresh(): void {
+  async refresh(): Promise<void> {
+    this.environments = await getEnvironmentsForBranch();
```

Comment:
what does it mean when you call to get Branch inside `refresh`?

Reply:
okay I see, you keep referring until you get response from github.

Reply:
until now the refresh function fired onDidChangeTreeData event and then we fetched the environments in the getChildren function, 
in order to know when to show and hide the loading message i moved the fetch of environments to the refresh function and i use it [here](https://github.com/env0/vscode-env0/blob/a90ef79430038e78ee0ace63a0fbfad4c4f716a8/src/extension.ts#L81) to know when refresh is done and i can hide the loading message.
the refresh happens once in the activate then every 3 s in the interval  and also after commands such as redeploy etc.

Reply:
I meant `refreshing` sorry for the typo